src/spikeinterface/postprocessing/spike_locations.py

```diff
@@ -59,9 +59,6 @@ class ComputeSpikeLocations(AnalyzerExtension):
     def __init__(self, sorting_analyzer):
         AnalyzerExtension.__init__(self, sorting_analyzer)
 
-        extremum_channel_inds = get_template_extremum_channel(self.sorting_analyzer, outputs="index")
-        self.spikes = self.sorting_analyzer.sorting.to_spike_vector(extremum_channel_inds=extremum_channel_inds)
```

Comment:
Unfortunatly this is a need!
The spike locations algo need the channel_index per spikes which is not in the "standard/basic" spike_vector.
We need the extremum_channel_inds to assign a channel_index per spikes which is needed exactly here
https://github.com/SpikeInterface/spikeinterface/blob/main/src/spikeinterface/postprocessing/spike_amplitudes.py#L184

Reply:
Merci.

We get rid of those lines, but the extremum_channel_inds are still calculated here:
https://github.com/SpikeInterface/spikeinterface/blob/17ee78552ddff147e1aaefcac70364c4a8520d99/src/spikeinterface/postprocessing/spike_locations.py#L103
and then sent to thepipeline on the line after that one. It's just that now they aren't calculated in the __init__, they've only calculated when the calculation happens.

I don't think spike_locations uses SpikeAmplitudeNode, which you've linked to.